code/modules/multiz/structures.dm

```diff
@@ -0,0 +1,233 @@
+
+	if(do_after(M, src, climb_time))
+		climbLadder(M, target_ladder)
+
+/obj/structure/ladder/attack_ghost(var/mob/M)
+	var/target_ladder = getTargetLadder(M)
+	if(target_ladder)
+		M.forceMove(get_turf(target_ladder))
+
+/obj/structure/ladder/proc/getTargetLadder(var/mob/M)
+	if((!target_up && !target_down) || (target_up && !istype(target_up.loc, /turf) || (target_down && !istype(target_down.loc,/turf))))
+		to_chat(M, "<span class='notice'>\The [src] is incomplete and can't be climbed.</span>")
+		return
+	if(target_down && target_up)
+		var/direction = alert(M,"Do you want to go up or down?", "Ladder", "Up", "Down", "Cancel")
+
+		if(direction == "Cancel")
+			return
+
+		if(!M.may_climb_ladders(src))
+			return
+
+		switch(direction)
+			if("Up")
+				return target_up
+			if("Down")
+				return target_down
+	else
+		return target_down || target_up
+
+/mob/proc/may_climb_ladders(var/ladder)
+	if(!Adjacent(ladder))
+		to_chat(src, "<span class='warning'>You need to be next to \the [ladder] to start climbing.</span>")
+		return FALSE
+	if(incapacitated())
+		to_chat(src, "<span class='warning'>You are physically unable to climb \the [ladder].</span>")
+		return FALSE
+	return TRUE
+
+/mob/observer/ghost/may_climb_ladders(var/ladder)
+	return TRUE
+
+/obj/structure/ladder/proc/climbLadder(var/mob/M, var/target_ladder)
+	var/turf/T = get_turf(target_ladder)
+	for(var/atom/A in T)
+		if(!A.CanPass(M, M.loc, 1.5, 0))
+			to_chat(M, "<span class='notice'>\The [A] is blocking \the [src].</span>")
+			return FALSE
+	return M.Move(T)
+
+/obj/structure/ladder/CanPass(obj/mover, turf/source, height, airflow)
+	return airflow || !density
+
+/obj/structure/ladder/update_icon()
+	icon_state = "ladder[!!(allowed_directions & UP)][!!(allowed_directions & DOWN)]"
+
+/obj/structure/ladder/up
+	allowed_directions = UP
+	icon_state = "ladder10"
+
+/obj/structure/ladder/updown
+	allowed_directions = UP|DOWN
+	icon_state = "ladder11"
+
+/obj/structure/stairs
+	name = "Stairs"
+	desc = "Stairs leading to another deck.  Not too useful if the gravity goes out."
+	icon = 'icons/obj/stairs.dmi'
+	icon_state = "stairs"
+	density = 0
+	opacity = 0
+	anchored = 1
+
+/obj/structure/stairs/initialize()
+	for(var/turf/turf in locs)
+		var/turf/simulated/open/above = GetAbove(turf)
+		if(!above)
+			warning("Stair created without level above: ([loc.x], [loc.y], [loc.z])")
+			return qdel(src)
+		if(!istype(above))
+			above.ChangeTurf(/turf/simulated/open)
+
+/obj/structure/stairs/Uncross(atom/movable/A)
+	if(A.dir == dir)
+		var/turf/simulated/open/above = GetAbove(A)
+		if(!above || !istype(above))
+			return
+		// This is hackish, but if we use normal forceMove() it ignores things on the turf above and in front in the way.
+		var/turf/target = get_step(above, dir)
+		var/turf/source = A.loc
+		if(target.Enter(A, source))
+			A.loc = target
+			target.Entered(A, source)
```

Comment:
Ditto

Reply:
Does not work if called like that

Reply:
Move() wasn't working, so I made it forceMove after the Enter() check and now it works

Reply:
Now works, resolved